File: client/java/src/main/java/com/be/mesh/client/system/MeshPublisher.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.system;

import com.be.mesh.client.annotate.SPI;
import com.be.mesh.client.mpc.ServiceProxy;
import com.be.mesh.client.prsim.Publisher;
import com.be.mesh.client.struct.Event;

import java.util.List;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public class MeshPublisher implements Publisher {

    private final Publisher publisher = ServiceProxy.proxy(Publisher.class);

    @Override
    public List<String> publish(List<Event> event) {
        return publisher.publish(event);
    }

    @Override
    public List<String> broadcast(List<Event> event) {
        return publisher.broadcast(event);
    }
}

File: client/golang/system/mesh_publisher.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package system

import (
	"context"
	"github.com/opendatav/mesh/client/golang/macro"
	"github.com/opendatav/mesh/client/golang/prsim"
	"github.com/opendatav/mesh/client/golang/types"
)

func init() {
	var _ prsim.Publisher = new(systemPublisher)
	macro.Provide(prsim.IPublisher, new(systemPublisher))
}

type systemPublisher struct {
}

func (that *systemPublisher) Att() *macro.Att {
	return &macro.Att{Name: macro.MeshSys}
}

func (that *systemPublisher) Publish(ctx context.Context, events []*types.Event) ([]string, error) {
	return aware.Publisher.Publish(ctx, events)
}

func (that *systemPublisher) Broadcast(ctx context.Context, events []*types.Event) ([]string, error) {
	return aware.Publisher.Broadcast(ctx, events)
}
